python/ray/autoscaler/v2/autoscaler.py

```diff
@@ -90,7 +93,13 @@ def _init_cloud_provider(
 
         """
         provider_config = config.get_provider_config()
-        if config.provider == Provider.READ_ONLY:
+        if provider_config["type"] == "kuberay":
```

Comment:
I find it a bit dangerous that we use a lot of hard-coded strings here.

Reply:
i agree - we could iterate on this, there are way TOO many in v1 and in v2 now that's not using a structured config schema, I think the `AutoscalingConfig` class should be a dataclass rather than a dict with getters.